app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -6,27 +6,30 @@
 
 class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
   include AuthenticatesWithTwoFactor
+  USER_TYPES = [HMIS_USER_TYPE = 'hmis_user'.freeze, WH_USER_TYPE = 'user'.freeze].freeze
 
   def okta
```

Comment:
If we're calling this endpoint from the HMIS, we also need to add it to the rack cors config https://github.com/greenriver/hmis-warehouse/blob/release-70/config/initializers/cors.rb. Currently, we only allow requests to `/hmis/*` endpoints from the HMIS. I think it is safest to call out the specific endpoints, so `/users/auth/okta` instead of `/users/auth`.

For the reverse-proxying to work, you'll also need to make a copy of this block https://github.com/greenriver/hmis-frontend/blob/13f0af85c014f813e0d17baa898bb72c029aadf4/docker/templates/default.conf.template#L26-L36 for the new locations. Again, I think it's probably safest to proxy only the specific routes that we need.

Reply:
Thanks @gigxz! I've added the auth end point to the cors config. Good point about making the the endpoint more specific. I'll adjust the the nginx config template for the hmis-frontend too.